docs/rules/prefer-destructuring.md

```diff
@@ -62,6 +62,45 @@ An example configuration, with the defaults filled in, looks like this:
 }
 ```
 
+Alternatively, you can use a different set of configuration objects to have more control on what to enforce destructuring. It also takes two sets of configuration objects.
+
+The first has two properties, `VariableDeclarator` and `AssignmentExpression`, which can be used to turn on or off the destructuring requirement for each of those types independently. Each property accepts an object that accpets two properties, `array` and `object`, which can be used to turn on or off the destructuring requirement for each of those types independently for variable declarations and assignment expressions.  By default, `array` and `object` are set to true for both `VariableDeclarator` and `AssignmentExpression`.
+
+The second has a single property, `enforceForRenamedProperties`, that controls whether or not the `object` destructuring rules are applied in cases where the variable requires the property being access to be renamed.
+
+Examples of **incorrect** code when `enforceForRenamedProperties` is enabled:
+
+```javascript
+var foo = object.bar;
+```
+
+Examples of **correct** code when `enforceForRenamedProperties` is enabled:
+
+```javascript
+var { bar: foo } = object;
+```
+
+An example configuration, with the defaults filled in, looks like this:
+
+```json
+{
+  "rules": {
+    "prefer-destructuring": ["error", {
+      "VariableDeclaration": {
```

Comment:
I like VariableDeclarator because I feel it's a closer parallel to AssignmentExpression, but I can see the potential for confusion. No strong opinion from me, though.